src/twisted/internet/defer.py

```diff
@@ -1095,14 +1099,19 @@ def main(reactor):
 
         @raise ValueError: If C{coro} is not a coroutine or generator.
         """
-        if not iscoroutine(coro) and not isinstance(coro, GeneratorType):
+        # type note: Subclass of "Generator[Deferred[_T], object, _T]" and "GeneratorType" cannot exist
+        if not iscoroutine(coro) and not isinstance(coro, GeneratorType):  # type: ignore[unreachable]
```

Comment:
I think this is an oddity of `GeneratorType` and `Generator` not being the same.

Reply:
There is a partial overlap: at least `Generator[T, None, None]` is accepted as an instance of `GeneratorType`. I found that out while working on Stan's annotations.

Reply:
I made a minimal test case from this and reported it to mypy: python/mypy#10501

Reply:
It turns out to be an issue in typeshed and a fix was already being worked on: python/typeshed#5496

Reply:
Cool; we'll remove the type hints when we bump to a Mypy with that fix.